### ISSUE TITLE
Support returning NO alignments when max-alignments is exceeded

### ISSUE THREAD

**Issue by eblume** [eblume/Tiger-Lily#10]
This ticket is in reference to **tigerlily.index.fixedtree.FixedTree.alignments**

There should be an option to enable 'cutoff mode', in which if maximum_alignments is exceeded NO alignments will be returned, rather than just the already gathered alignments. This would be best implemented by having the tree searching algorithm raise an exception when maximum_alignments is exceeded. This cutoff mode should be **enabled by default**.

Additionally there is currently a related bug: if maximum_alignments is specified the logic currently may return MORE than maximum_alignments if it happens to find more than one alignment at the current search depth. This should not be allowed to happen. Because it is computationally very easy to gather the extra alignments, we should still have the search function find the possibly-too-large set of alignments and then just lop off the extra alignments.

I agree that this is suboptimal behavior in terms of performance, however consider two things:
1. It is _very bad_ to return N+1 (or more) alignments when maximum_alignments is set to N
2. I imagine that most applications will leave the cutoff mode on, in which case the point is moot as we won't return _any_ alignments

**Comment by eblume** [comment]
This is being put on hold for the short term because based on some performance tests it looks like this class will need to be mostly rewritten in C - python can't handle the string manipulation and table lookups at such a large scale.

**Comment by eblume** [comment]
Blocked by #18
